Sources/edge/Commands/ImagesCommand.swift

```diff
@@ -0,0 +1,179 @@
+    let nextPageToken: String?
+}
+
+// MARK: - GCS Browser
+
+/// Browser for a publicly-readable Google Cloud Storage bucket.
+actor GCSBucketBrowser {
+    private let session: URLSession
+    private let bucket: String
+
+    init(bucket: String, session: URLSession = .shared) {
+        self.bucket = bucket
+        self.session = session
+    }
+
+    /// Lists up to `maxResults` object names (with optional prefix filtering).
+    /// - Parameters:
+    ///   - prefix: only return objects whose names begin with this string
+    ///   - pageToken: if non-nil, continue listing from this token
+    ///   - maxResults: maximum number of results to return (1-1000)
+    func listObjects(prefix: String? = nil,
+                     pageToken: String? = nil,
+                     maxResults: Int = 1000) async throws -> GCSObjectList
+    {
+        var components = URLComponents()
+        components.scheme = "https"
+        components.host = "www.googleapis.com"
+        components.path = "/storage/v1/b/\(bucket)/o"
+        var queryItems: [URLQueryItem] = [
+            URLQueryItem(name: "maxResults", value: String(maxResults)),
+            // Only grab the fields we actually use
+            URLQueryItem(name: "fields", value: "items(name,size,contentType),nextPageToken")
+        ]
+        if let prefix {
+            queryItems.append(.init(name: "prefix", value: prefix))
+        }
+        if let pageToken {
+            queryItems.append(.init(name: "pageToken", value: pageToken))
+        }
+        components.queryItems = queryItems
+
+        guard let url = components.url else {
+            throw URLError(.badURL)
+        }
+
+        let (data, response) = try await session.data(from: url)
+        guard let http = response as? HTTPURLResponse,
+              (200...299).contains(http.statusCode)
+        else {
+            throw URLError(.badServerResponse)
+        }
+
+        return try JSONDecoder().decode(GCSObjectList.self, from: data)
+    }
+
+    /// Walks all pages and yields every object name via an async stream.
+    func allObjectNames(prefix: String? = nil) -> AsyncThrowingStream<String, Error> {
+        AsyncThrowingStream { continuation in
+            Task {
+                var token: String? = nil
+                repeat {
+                    let page = try await listObjects(prefix: prefix, pageToken: token)
+                    page.items?.forEach { continuation.yield($0.name) }
+                    token = page.nextPageToken
+                } while token != nil
+                continuation.finish()
+            }
+        }
+    }
+}
+
+struct ImagesCommand: AsyncParsableCommand {
+    static let configuration = CommandConfiguration(
+        commandName: "images",
+        abstract: "List available edge-agent images in the storage bucket"
+    )
+    
+    @Option(name: .long, help: "Storage bucket name (default: apache-edge-agent-releases)")
+    var bucket: String = "edgeos-images-public"
```

Comment:
The default isn't what it says in the descritption

Reply:
yep. I need to pick this up it's not finished